src/rgw/rgw_rest.cc

```diff
@@ -161,20 +162,21 @@ string lowercase_underscore_http_attr(const string& orig)
  */
 string uppercase_underscore_http_attr(const string& orig)
 {
-  const char *s = orig.c_str();
-  char buf[orig.size() + 1];
-  buf[orig.size()] = '\0';
-
-  for (size_t i = 0; i < orig.size(); ++i, ++s) {
-    switch (*s) {
-      case '-':
-        buf[i] = '_';
-        break;
-      default:
-        buf[i] = toupper(*s);
-    }
-  }
-  return string(buf);
+  std::string mod;
+  mod.reserve(orig.size());
+  std::transform(orig.cbegin(),
+		 orig.cend(),
+		 std::back_inserter(mod),
```

Comment:
Since this looks like the same switch-statement, consider making it its own lil' function and sharing it.